board.h

```diff
@@ -0,0 +1,71 @@
+#include <vector>
+#include <iostream>
+#include <string>
+#include <cstdlib> 
+
+struct Position{
+    int xpos;
+    int ypos;
+
+    bool operator==(const Position b) const{
+        return ((xpos == b.xpos)
+            && (ypos == b.ypos));
+    }
+    Position(){
+
+    }
+    Position(int x, int y){
+        this->xpos = x;
+        this->ypos = y;
+    }
+};
+
+struct Mine : Position{

```

Comment:
#must not use inheritance for this, use composition. Inheritance implies an is-a relationship, and a `Mine` is NOT a `Position`; the `Mine` HAS a `Position`.